Visitor can read a specific article CLIENT

FlisAnn opened the pull request:
[Visitor can read a specific article CLIENT](https://www.pivotaltracker.com/story/show/176053423)
### User Story
```
As a Visitor 
In order to read an article 
I would like to be able to click on an article and have it displayed
```
## Changes proposed in this pull request:
* Visitor can click an article and come to that page
* Visitor can see a header with links to different categories
* Visitor can see a footer
* Visitor is presented with nice error messages if there are no articles or if a specific article can't be displayed
* Refactoring getArticle module

## What I have learned working on this feature:
* How to work with paths between article list and article
* How  to use one module to perform multiple axios actions

## Packages/Gems added
* react-router-dom

## Screenshots (Client)
![Screenshot 2020-12-13 at 11 33 38](https://user-images.githubusercontent.com/72142651/102009548-682a5600-3d38-11eb-87dc-f60549b9c9e8.png)
![Screenshot 2020-12-13 at 11 35 20](https://user-images.githubusercontent.com/72142651/102009552-6e203700-3d38-11eb-841f-10c9a2e27a3c.png)

FlisAnn committed:
Adding first test for clicking on a article

FlisAnn committed:
Test finds first title

FlisAnn committed:
Working on creating link for first article

psnoya committed:
Added a route path

psnoya committed:
ping pong

FlisAnn committed:
Merge branch 'visitor_can_read_a_specific_article' of https://github.com/psnoya/kitty_news_client_user into visitor_can_read_a_specific_article

FlisAnn committed:
Adding browserrouter and NavLink

sealfury committed:
Adjusting naming conventions in test/component
Providing basic framework for hooks & content in ShowArticle component

psnoya committed:
Adding useEffect hook

FlisAnn committed:
Importing
useEFfect, useSTate, showArticle in DisplyArticle

Ehugo2000 committed:
Trying to fix routing error for specific article.

FlisAnn committed:
Refactoring Display-articlesList

FlisAnn committed:
Fixing test so that a article can be clicked

salindholm committed:
Refactoring getArticles module. Modifying imports.

salindholm committed:
Correcting test to follow sequence, hardcoded title to check to test test functionallity

salindholm committed:
Changing import of DisplayArticlesList in app.jsx and changing name of const.

FlisAnn committed:
Fixing typo in fixture file and thus the test are passing and a user can now see a specific article

FlisAnn committed:
Formating the code to look pretty

FlisAnn committed:
Adding test for lead and body, passing green

FlisAnn committed:
Fixing text in test to be more understandable

FlisAnn committed:
Creating sad path test for an article not showing correctly

FlisAnn committed:
Adding a working test for sad path: If article is broken user will see a message

salindholm committed:
Refactoring useEffect to work properly

salindholm committed:
Creating a header component with a navbar for home news and some dummy categories

salindholm committed:
Creating a test for user to navigate from article back to news page

salindholm committed:
Importing Header to app.jsx

salindholm committed:
Removing data-cy from segment and modyfing test.

FlisAnn committed:
Creating test for footer

FlisAnn committed:
Creating Footer.jsx

FlisAnn committed:
Adding styling to footer

FlisAnn committed:
Adding containers for error messages

emtalen commented on cypress/integration/userCanNavigateToFooter.feature.js:
```diff
@@ -0,0 +1,14 @@
+  it("see a footer", () => {
+    cy.get("[data-cy='footer']").should("exist");
+  });
+});
```
This test is unnecessary

psnoya replied:
Has been deleted.

emtalen commented on cypress/integration/userCanNavigateToHeader.feature.js:
```diff
@@ -0,0 +1,29 @@
+      cy.get("[data-cy='article-index']").should("exist")
+		});
+	});
+});
```
this test is also unnecessary, you should only test logic and functionality.

psnoya replied:
We have removed this now.

emtalen commented on cypress/integration/userCanReadASpecificArticle.feature.js:
```diff
@@ -0,0 +1,62 @@
+      cy.server();
+      cy.route({
+        method: "GET",
+        url: "http://localhost:3000/api/articles/abc",
```
for readability of the test you should probably make the request with an id that you don't have but is still a number. The use case for this is example if an article got deleted the same second I pressed on it. 
Why you used `abc` in your backend is simply to trigger the error and make sure that it also catches if someone would make that request from another resource than your client

psnoya replied:
We changed it to 123456.

emtalen commented on src/components/DisplayArticle.jsx:
```diff
@@ -0,0 +1,37 @@
+      ) : (
+        <Container data-cy="error-article">
+          <h1 >
+            Something went wrong, this article was not found
```
You are sending a message from your backend, please make use of that instead of hard coding it here

emtalen commented on src/components/Header.jsx:
```diff
@@ -0,0 +1,38 @@
+	);
+};
+
+export default Header;
```
please format this file and make sure that to use indentation using spaces instead of tabs. 
Also, I think you should definitely have a header, but adding a lot of buttons that leads to the same place is not very user friendly. Please stick to one button that can tak the user back to all the articles until you add categories

psnoya replied:
Fixed it up!

emtalen reviewed (commented):
Really good job on this one team!! ✨  I'm really proud that you nailed this functionality with the params hook in browser router.

psnoya committed:
Removing userCanNavigateFoorer.feature.js
Removing verification of article-index in userCanNavigateHeader.feature.js
Formatting file and removing dummy-links

emtalen committed:
reafactoring to make use of the error message coming from the backend

emtalen commented on cypress/integration/userCanNavigateToHeader.feature.js:
```diff
@@ -0,0 +1,28 @@
+      cy.get("[data-cy='header-news']").click();
+		});
+	});
+});
```
This is still here...

emtalen commented on src/modules/getArticles.js:
```diff
@@ -1,8 +1,19 @@
 
-export { getArticles };
\ No newline at end of file
+export default getArticles;
```
So modules should actually be exported in brackets, not default, so change that back 👍🏻

emtalen reviewed (commented):
Just small things left now 🙏🏻

salindholm committed:
Deleting test for navigation with header and correcting export and imports of getArticles module

emtalen committed:
removing x before the it block that i forgot
